tests/foreman/ui/test_products.py

```diff
@@ -157,12 +143,14 @@ def test_positive_update_basic(self, new_prd_name):
                 description=gen_string('alphanumeric'),
             )
             self.assertIsNotNone(self.products.search(prd_name))
-            self.products.update(prd_name, new_name=new_prd_name)
-            self.assertIsNotNone(self.products.search(new_prd_name))
+            for new_prd_name in generate_strings_list():
+                with self.subTest(new_prd_name):
+                    self.products.update(prd_name, new_name=new_prd_name)
+                    self.assertIsNotNone(self.products.search(new_prd_name))
+                    prd_name = new_prd_name  # for next iteration
 
     @run_only_on('sat')
-    @data(*generate_strings_list())
-    def test_positive_update_to_original_name(self, prd_name):
+    def test_positive_update_to_original_name(self):
```

Comment:
I intentionally changed this from datadriven to normal test as repeating this test for all generate_strings_list() is redundant.